source/endpoints_ruleset.c

```diff
@@ -467,32 +463,22 @@ static int s_on_parameter_key(
     if (deprecated_node != NULL) {
         struct aws_json_value *deprecated_message_node =
             aws_json_value_get_from_object(deprecated_node, aws_byte_cursor_from_c_str("message"));
-        if (deprecated_message_node != NULL) {
-            struct aws_byte_cursor deprecated_message_cur;
-            if (aws_json_value_get_string(deprecated_message_node, &deprecated_message_cur)) {
-                AWS_LOGF_ERROR(AWS_LS_SDKUTILS_ENDPOINTS_PARSING, "Unexpected value for deprecated message.");
-                goto on_error;
-            }
-            parameter->deprecated_message = aws_string_new_from_cursor(wrapper->allocator, &deprecated_message_cur);
+        if (deprecated_message_node != NULL &&
+            aws_json_value_get_string(deprecated_message_node, &parameter->deprecated_message)) {
+            AWS_LOGF_ERROR(AWS_LS_SDKUTILS_ENDPOINTS_PARSING, "Unexpected value for deprecated message.");
+            goto on_error;
         }
 
         struct aws_json_value *deprecated_since_node =
             aws_json_value_get_from_object(deprecated_node, aws_byte_cursor_from_c_str("since"));
-        if (deprecated_since_node != NULL) {
-            struct aws_byte_cursor deprecated_since_cur;
-            if (aws_json_value_get_string(deprecated_since_node, &deprecated_since_cur)) {
-                AWS_LOGF_ERROR(AWS_LS_SDKUTILS_ENDPOINTS_PARSING, "Unexpected value for deprecated since.");
-                goto on_error;
-            }
-            parameter->deprecated_since = aws_string_new_from_cursor(wrapper->allocator, &deprecated_since_cur);
+        if (deprecated_since_node != NULL &&
+            aws_json_value_get_string(deprecated_since_node, &parameter->deprecated_since)) {
+            AWS_LOGF_ERROR(AWS_LS_SDKUTILS_ENDPOINTS_PARSING, "Unexpected value for deprecated since.");
+            goto on_error;
         }
-    } else {
-        parameter->is_deprecated = false;
-        parameter->deprecated_message = NULL;
-        parameter->deprecated_since = NULL;
     }
 
-    aws_hash_table_put(wrapper->table, &parameter->name_cur, parameter, NULL);
+    aws_hash_table_put(wrapper->table, &parameter->name, parameter, NULL);
```

Comment:
trivial: this can fail, but it seems like only fail with OOM, and we don't handle again... Just bring it up, maybe ASSERT on it to be succeed?